consensus/types/src/validator.rs

```diff
@@ -98,6 +100,35 @@ impl Validator {
             .flatten()
     }
 
+    /// New in electra
+    pub fn has_compounding_withdrawal_credential(&self, spec: &ChainSpec) -> bool {
+        is_compounding_withdrawal_credential(&self.withdrawal_credentials, spec)
+    }
+
+    // New in electra
+    pub fn has_execution_withdrawal_credential(&self, spec: &ChainSpec) -> bool {
+        self.has_compounding_withdrawal_credential(spec)
+            || self.has_eth1_withdrawal_credential(spec)
+    }
+
+    pub fn get_max_effective_balance(&self, spec: &ChainSpec) -> u64 {
+        if self.has_compounding_withdrawal_credential(spec) {
+            spec.max_effective_balance_eip7251
+        } else {
+            // FIXME: does this need to be made fork-aware?
```

Comment:
Depending on where it's used, yes. 0x02 validators before the fork will behave as if they had 32 ETH maxeb